### PR TITLE
Contract version approach shared

### PR DESCRIPTION
**Pull request by iainnash**
All contracts get the same static version.

### CODE REVIEW

File: src/VersionedContract.sol [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.16;
+
+abstract contract VersionedContract {
+    function contractVersion() external returns (string memory) {
+        return "1.0.2";
```
Comment (tbtstl) [inline src/VersionedContract.sol:6]: we're going to make this `1.0.0`, right?

File: src/manager/Manager.sol [hunk 2]
```diff
@@ -13,10 +13,13 @@ import { IAuction } from "../auction/IAuction.sol";
...
 import { IGovernor } from "../governance/governor/IGovernor.sol";
 
+import { VersionedContract } from "../VersionedContract.sol";
+
 /// @title Manager
 /// @author Rohan Kulkarni
+/// @custom:repo github.com/ourzora/nouns-protocol 
 /// @notice The DAO deployer and upgrade manager
-contract Manager is IManager, UUPS, Ownable, ManagerStorageV1 {
+contract Manager is VersionedContract, IManager, UUPS, Ownable, ManagerStorageV1 {
```
Comment (tbtstl) [inline src/manager/Manager.sol:22]: nit: consistency in importing interface `IManager` before versioned contract?

(see above, in `Treasury.sol`)

**Review by tbtstl — APPROVED** [review]
A couple of nits, but excited for this one to go out.

File: src/VersionedContract.sol (hunk 1 above)
Comment (iainnash) [reply src/VersionedContract.sol:6]: Yeah, probably a good idea.

**Pull request merged**